### ISSUE TITLE
Auto-updater is bad

### ISSUE THREAD

**Issue by Cadiboo** [Cadiboo/NoCubes#19]
It downloads the entire jar. Make a patch-based version as a seperate library

**Comment by Rocketblaster247** [comment]
Yes, please remove the auto-updater, or at least make it an option.

**Comment by Cadiboo** [comment]
Yeah will do in the rewrite. Any reason why you want it disabled? I would think that automatically getting bug fixes and updates would be good

**Comment by Rocketblaster247** [comment]
Latest 1.12 release wasn't always working on my PC. I would just use the release before it. The latest 1.12 release seems to be working now. Thanks and sorry for taking so long!

**Comment by Cadiboo** [comment]
Interesting, did you report the bug anywhere? Auto updater is disableable in the 1.15 rewrite.

**Comment by Rocketblaster247** [comment]
It's fine. It seems to be working now! THX

**Comment by Creeper36** [comment]
I am running 1.12.2 and it keeps downloading 0.4.0-pre1 but I am already using NoCubes-1.12.2-0.3.3.jar.. it errors out my launcher until I manually delete this file.  you have the wrong version listed in the update link.

== FORGE ERROR ==
[Forge Version Check/INFO] [forge.VersionCheck]: [nocubes] Starting version check at https://Cadiboo.github.io/projects/nocubes/update.json

== RESULTS OF UPDATE.JSON VIA BROWSER ==
{
	"homepage": "https://Cadiboo.github.io/projects/nocubes/",
	"promos": {
		"1.14.4-recommended": "1.14.4-0.4.0-pre8",
		"1.14.4-latest": "1.14.4-0.4.0-pre8",
		"1.14.3-latest": "1.14.3-0.3.0-pre5",
		"1.14.2-latest": "1.14.2-0.3.0-pre1",
		"1.13.2-latest": "1.13.2-0.3.0",
		"1.13.2-recommended": "1.13.2-0.3.0",
		"1.12.2-latest": "1.12.2-0.4.0-pre1",
		"1.12.2-recommended": "1.12.2-0.4.0-pre1"
	}
}

**Comment by Creeper36** [comment]
when using 'NoCubes-1.12.2-0.4.0-pre1.jar' I keep getting 'item_owl_egg' error.. but I do Not get this error and Forge loads fine with NoCubes-1.12.2-0.3.3.. but I keep getting the update and its causing a crash cause I have both versions in my mods folder.. so I edited 'mcmod.info' to this: 

LINE 9:  	"updateJSON": "",

Now I get a working version.. no errors or crashes or duplicate files.. can you please update this committ and reupload to sourceforge for others?  Thanks

**Comment by Creeper36** [comment]
I'm using Forge (Minecraft 1.12.2) with Optifine.. I have my own server and run 440 mods.. I get a crash of "item_owl_egg" while using NoCubes-1.12.2-0.4.0-pre1.jar but I do Not get a crash with NoCubes-1.12.2-0.3.3.jar but due to the auto-updater I have to delete the newer version that autodownloads.. if you have a 0.3.3 version with this disabled then it would save me some headaches each time me and my friends load this game.

**Comment by Cadiboo** [comment]
> Auto updater is disableable in the 1.15 rewrite.

Alternatively you can turn off Forge’s update checker that NoCubes uses in the config
